### PR TITLE
feat: Add Pro components and Examples

### PR DESCRIPTION
**Pull request by wss-rbrennan**
Add Pro components and examples.

- reorganizes templates by resources type ex.  deployment-, service-, hpa-
- add environment variable templates for modules
- add helpers for pro, standard, and common names
- add example for standard without dependencies
- add example for standard version with dependencies
- add example for professional version without dependencies
- add example for professional version with all traditional dependencies (clickhouse, kafka, zookeeper, mongodb, redis)
- add example for professional version replacing zookeeper with kafka kraft mode, and clickhouse keeper.
- add the "charts" folder and dependency charts for easier first time deployment

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,6 @@
 .vscode
 .project
 .settings
-charts/*/charts
```
Comment (dsun0720) [inline .gitignore]: I've created a GitHub action that will automatically package dependencies and publish this chart. No need to upload these dependence's packages

**Review by dsun0720 — CHANGES_REQUESTED** [review]
Thank you very much for your contribution to the charts in the PRO section.

In your PR, you can add new YAML or TPL files without modifying the file names of existing files. This is because we already have a new branch in place based in v0.0.1 in order to optimize the auto discovery of externe IPs and backup mongoDB.

I hope you can understand.

File: .gitignore (hunk 1 above)
Comment (wss-rbrennan) [reply .gitignore]: Yeah, I can change the file names back to their original names.  I'd like to recommend that post v0.0.2 that the names be grouped by resource type.

Comment (wss-rbrennan) [reply .gitignore]: fixed with commit # d69a6e3

**Review by dsun0720 — CHANGES_REQUESTED** [review]
截屏2023-07-08 09 22 12


can you add this limit in .github/release.yaml ? I'll publish a new release with your pr and v0.0.2. 

Thank you

**Comment by wss-rbrennan** [comment]
> 截屏2023-07-08 09 22 12
> 
> can you add this limit in .github/release.yaml ? I'll publish a new release with your pr and v0.0.2.
> 
> Thank you

Added with commit # 17b8cae

**Review by dsun0720 — APPROVED** [review]

**Pull request merged**